src/js/common/schemaform/FormApp.jsx

```diff
@@ -40,6 +40,16 @@ class FormApp extends React.Component {
     if (window.History) {
       window.History.scrollRestoration = 'manual';
     }
+
+    // If we start in the middle of a form, redirect to the beginning
+    const currentPath = this.props.location.pathname;
```

Comment:
I think you may want to use `currentLocation` instead of `location`. That's what the other code in this component uses. Or we could probably switch to using `location` and remove the `currentLocation` prop.